tripy/tests/integration/test_plugin_qdp.py

```diff
@@ -0,0 +1,184 @@
+    inp0: trtp.TensorDesc,
+    outputs: Tuple[trtp.TensorDesc],
+) -> List[trtp.AutoTuneCombination]:
+
+    return [trtp.AutoTuneCombination("FP32|FP16, FP32|FP16")]
+
+@trtp.aot_impl("example::circ_pad_plugin")
+def circ_pad_plugin_aot_impl(
+    inp0: trtp.TensorDesc, pads: npt.NDArray[np.int32], outputs: Tuple[trtp.TensorDesc], tactic: int
+) -> Tuple[Union[str, bytes], Union[str, bytes], trtp.KernelLaunchParams, trtp.SymExprs]:
+
+    block_size = 256
+
+    N = inp0.ndim
+    all_pads = np.zeros((N * 2,), dtype=np.int32)
+    inp_dims = inp0.shape_expr
+    out_dims = outputs[0].shape_expr
+
+    for i in range(np.size(pads) // 2):
+        all_pads[N * 2 - 2 * i - 2] = pads[i * 2]
+        all_pads[N * 2 - 2 * i - 1] = pads[i * 2 + 1]
+
+    all_pads = all_pads.tolist()
+    extra_args = trtp.SymIntExprs.from_tuple(
+        [
+            trtp.SymInt32(e)
+            for e in [
+                all_pads[0],
+                all_pads[2],
+                all_pads[4],
+                all_pads[6],
+                inp_dims[0],
+                inp_dims[1],
+                inp_dims[2],
+                inp_dims[3],
+                out_dims[1],
+                out_dims[2],
+                out_dims[3],
+                inp_dims.numel(),
+                out_dims.numel(),
+            ]
+        ]
+    )
+
+    type_str = "fp32" if inp0.dtype == trt.float32 else "fp16"
+
+    src = triton.compiler.ASTSource(
+        fn=circ_pad_kernel,
+        signature=f"*{type_str},{','.join(['i32']*13)},*{type_str}",
+        constants={
+            "BLOCK_SIZE": block_size,
+        },
+    )
+
+    compiled_kernel = triton.compile(src)
+    launch_params = trtp.KernelLaunchParams()
+
+    launch_params.grid_x = trtp.cdiv(out_dims.numel(), block_size)
+    launch_params.block_x = compiled_kernel.metadata.num_warps * 32
+    launch_params.shared_mem = compiled_kernel.metadata.shared
+
+    return compiled_kernel.metadata.name.encode(), compiled_kernel.asm["ptx"].encode(), launch_params, extra_args
+
+class TestQuickDeployablePlugin:
+    def test_elemwise_add_plugin(self):
+        inp = tp.iota((2, 2))
+        out = tp.plugin(
+            trt.plugin.op.example.elemwise_add_plugin,
+            [inp],
+            block_size = 256,
+        )
+
+        assert cp.allclose(cp.from_dlpack(out), cp.from_dlpack(inp + 1))
+
+    def test_circ_pad_plugin():
+        inp_shape = (10, 3, 32, 32)
+        x = np.random.normal(size=inp_shape).astype(trt.nptype(trt.DataType.FLOAT))
+
+        inp = tp.Tensor(x)
+        pads = np.array((1, 1, 1, 1), dtype=np.int32)
+
+        out = tp.plugin(
+            trt.plugin.op.example.circ_pad_plugin,
+            [inp],
+            pads = pads,
+        )
+
+        ref = np.pad(x, [[0, 0], [0, 0], [pads[0], pads[1]], [pads[2], pads[3]]], "wrap")
+
+        assert cp.allclose(cp.from_dlpack(out), ref)
+
+if __name__ == "__main__":
+    test_elemwise_add_plugin()
+    test_circ_pad_plugin()
```

Comment:
Can probably remove this part. You can do this via `pytest <path> -k <test_name>`